src/wp-login.php

```diff
@@ -997,14 +996,16 @@ function wp_login_viewport_meta() {
 		 * @param WP_User|WP_Error $user   WP_User object if the login and reset key match. WP_Error object otherwise.
 		 */
 		do_action( 'validate_password_reset', $errors, $user );
-
+		if ( isset( $_GET['user'] ) ) {
+			$username = wp_unslash( $_GET['user'] );
+		}
 		if ( ( ! $errors->has_errors() ) && isset( $_POST['pass1'] ) && ! empty( $_POST['pass1'] ) ) {
 			reset_password( $user, $_POST['pass1'] );
 			setcookie( $rp_cookie, ' ', time() - YEAR_IN_SECONDS, $rp_path, COOKIE_DOMAIN, is_ssl(), true );
 			login_header(
 				__( 'Password Reset' ),
 				wp_get_admin_notice(
-					__( 'Your password has been reset.' ) . ' <a href="' . esc_url( wp_login_url() ) . '">' . __( 'Log in' ) . '</a>',
+					__( 'Your password has been reset.' ) . ' <a href="' . esc_url( wp_login_url() ) . '?' . 'user=' . $username . '">' . __( 'Log in' ) . '</a>',
```

Comment:
This is not constricted in a safe way. It is vulnerable to an injection attack, since the `$username` is not getting escaped.

```suggestion
					__( 'Your password has been reset.' ) . ' <a href="' . esc_url( add_query_arg( 'user', rawurlencode( $username ), wp_login_url() ) ) . '">' . __( 'Log in' ) . '</a>',
```

Reply:
Something even better would be to obtain the login URL above, and append the `user` if the query var is present:

```php
$login_url = wp_login_url();
if ( isset( $_GET['user'] ) ) {
	$login_url = add_query_arg( 'user', rawurlencode( wp_unslash( $_GET['user'] ) ), $login_url );
}
```

And then here:


```suggestion
					__( 'Your password has been reset.' ) . ' <a href="' . esc_url( $login_url ) . '">' . __( 'Log in' ) . '</a>',
```